pkg/registry/apps/deployment/storage/storage.go

```diff
@@ -404,5 +418,12 @@ func (i *scaleUpdatedObjectInfo) UpdatedObject(ctx context.Context, oldObj runti
 	// move replicas/resourceVersion fields to object and return
 	deployment.Spec.Replicas = scale.Spec.Replicas
 	deployment.ResourceVersion = scale.ResourceVersion
+
+	updatedEntries, err := fieldmanager.MergeSubresourceEntries(deployment.ManagedFields, scale.ManagedFields, "apps/v1", "spec", "replicas")
```

Comment:
Same `deployment.MangedFields` here is passed again, and must ABSOLUTELY be the same as before. There is an implict state that should be preserved (including the `.spec.replicas` path) in a proper object.

Reply:
Totally, let me try and use your suggestion to organise the code a bit better.